flaml/tune/tune.py

```diff
@@ -33,16 +34,70 @@
 INCUMBENT_RESULT = "__incumbent_result__"
 
 
+def is_nan_or_inf(value):
+    return np.isnan(value) or np.isinf(value)
+
+
 class ExperimentAnalysis(EA):
     """Class for storing the experiment results."""
 
-    def __init__(self, trials, metric, mode):
+    def __init__(self, trials, metric, mode, lexico_objectives):
         try:
             super().__init__(self, None, trials, metric, mode)
         except (TypeError, ValueError):
             self.trials = trials
             self.default_metric = metric or DEFAULT_METRIC
-            self.default_mode = mode
+            self.default_mode = mode or DEFAULT_MODE
+            self.lexico_objectives = lexico_objectives

```

Comment:
This doesn't look right.
First, shouldn't lexico_objectives be optional?
Second, should line 51 be out of the except block? (not sure about line 50)
function get_best_trial() should get an error when `self.lexico_objectives` is undefined. Is that function covered in the test comprehensively?

Reply:
yes, I revised it now